frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/api/TopicResource.java

```diff
@@ -0,0 +1,169 @@
+
+/**
+ * Topic Resource executing commands through command executor.
+ * Kafka package should also be deployed with framework to enable Kafka commands.
+ */
+
+@Path("/v1/topics")
+public class TopicResource {
+    private static final Log log = LogFactory.getLog(TopicResource.class);
+
+    private final KafkaZKClient kafkaZkClient;
+    private final CmdExecutor cmdExecutor;
+
+    public TopicResource(CmdExecutor cmdExecutor, KafkaZKClient kafkaZkClient) {
+        this.kafkaZkClient = kafkaZkClient;
+        this.cmdExecutor = cmdExecutor;
+    }
+
+    @GET
+    public Response topics() {
+        try {
+            return Response.ok(kafkaZkClient.listTopics(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/{name}")
+    public Response getTopic(@PathParam("name") String topicName) {
+        try {
+            return Response.ok(kafkaZkClient.getTopic(topicName), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topic: " + topicName + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @POST
+    @Path("/create")
+    public Response createTopic(
+            @QueryParam("name") String name,
+            @QueryParam("partitions") String partitionCount,
+            @QueryParam("replication") String replicationFactor) {
+        try {
+            int partCount = Integer.parseInt(partitionCount);
+            int replFactor = Integer.parseInt(replicationFactor);
+            JSONObject result = cmdExecutor.createTopic(name, partCount, replFactor);
+            return Response.ok(result.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to create topic: " + name + " with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/unavailable_partitions")
+    public Response unavailablePartitions() {
+        try {
+            JSONObject obj = cmdExecutor.unavailablePartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @GET
+    @Path("/under_replicated_partitions")
+    public Response underReplicatedPartitions() {
+        try {
+            JSONObject obj = cmdExecutor.underReplicatedPartitions();
+            return Response.ok(obj.toString(), MediaType.APPLICATION_JSON).build();
+        } catch (Exception ex) {
+            log.error("Failed to fetch topics with exception: " + ex);
+            return Response.serverError().build();
+        }
+    }
+
+    @PUT
+    @Path("/operations/{type}")
```

Comment:
Now that I'm thinking about it, do you think it makes more sense to have this be at `/{topic}/operation/{type}`? That's what I'd go for intuitively if I were constructing URLs by hand. Also, this should probably be `POST`?